server/concepts/permissioning.ts

```diff
@@ -0,0 +1,65 @@
+import { ObjectId } from "mongodb";
+import DocCollection, { BaseDoc } from "../framework/doc";
+import { NotAllowedError, NotFoundError } from "./errors";
+
+export interface PermissionDoc extends BaseDoc {
+  user: ObjectId;
+}
+
+export default class PermissioningConcept {
+  public readonly organizers: DocCollection<PermissionDoc>;
+  public readonly members: DocCollection<PermissionDoc>;
+
+  constructor(collectionName1: string, collectionName2: string) {
+    this.organizers = new DocCollection<PermissionDoc>(collectionName1);
+    this.members = new DocCollection<PermissionDoc>(collectionName2);
+  }
+
+  async addMember(u: ObjectId) {
+    await this.assertAlreadyExists(u);
+    const _id = await this.members.createOne({ user: u });
+    return { msg: "Member successfully created!", member: await this.members.readOne({ _id }) };
+  }
+
+  async addOrganizer(u: ObjectId) {
+    await this.assertAlreadyExists(u);
```

Comment:
id remove line 25 bc if they dont exist as a member yet, this will throw error b4 u get to line 30

Reply:
or replace with a new function like assertOrganizerAlreadyExists, throw an AlreadyExistsError error if the id exists in organizers already